src/gui/codeeditors/qgscodeeditor.h

```diff
@@ -81,8 +85,11 @@ class GUI_EXPORT QgsCodeInterpreter
 };
 
 
-
-class QWidget;
+// TODO QGIS 4.0 -- make QgsCodeEditor inherit QWidget only,
```

Comment:
Would it make sense to go for a V2 and implement this directly?

Reply:
It's actually not trivial, and would require some quite extensive refactoring. I'd rather not :laughing:

Reply:
Actually, I've revised the TODO note to make it clear that this is a consideration only and not a must-do. I'm not convinced anymore that it would make the code any cleaner.